yarn-project/simulator/src/avm/avm_machine_state.ts

```diff
@@ -46,6 +46,7 @@ export class AvmMachineState {
   constructor(gasLeftOrL2GasLeft: Gas | number, daGasLeft?: number) {
     if (typeof gasLeftOrL2GasLeft === 'object') {
       ({ l2Gas: this.l2GasLeft, daGas: this.daGasLeft } = gasLeftOrL2GasLeft);
+      this.l2GasLeft -= FIXED_AVM_STARTUP_L2_GAS;
```

Comment:
Can this potentially make the gas left negative or undeflow?

Reply:
Yes! Fixed this.